packages/augur-ui/src/modules/swap/components/index.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState } from 'react';
+
+import { ETH, DAI, REP } from 'modules/common/constants';
+import { AccountBalances, FormattedNumber } from 'modules/types';
+import {
+  SwapArrow,
+  REP as REPIcon,
+  ETH as ETHIcon,
+  DaiLogoIcon,
+} from 'modules/common/icons';
+import { formatEther, formatDai } from 'utils/format-number';
+
+import Styles from 'modules/swap/components/index.styles.less';
+
+interface SwapProps {
+  balances: AccountBalances;
+  toToken: string;
+  fromToken: string;
+}
+
+export const Swap = ({ balances, fromToken, toToken }: SwapProps) => {
+  // TODO placeholder rates until price feed is hooked up
+  const ETH_RATE = 160.63;
+  const REP_RATE = 15.87;
+
+  const setFromToken = () => {
+    const nextToken = selectedToken === ETH ? DAI : ETH;
+    setSelectedFromTokenAmount('');
+    setSelectedToken(nextToken);
+  };
+
+  const setTokenAmount = (amount, displayBalance) => {
+    if (amount < 0) {
```

Comment:
Should be using BigNumber for everything. Tokens amounts can go out to 18 digits.

Reply:
Ah darn I meant to update that, fixing